### ISSUE TITLE
Pass names to cohens_d() / sd_pooled()

### ISSUE THREAD

**Issue by david-romano** [easystats/effectsize#98]
Hi, just an observation about what could be treated as an ambiguity in the documentation, or a bug if the documentation is intended literally:

argument | expected type of value
-- | --
x | A formula, a numeric vector, or a name of one in data.
y | A numeric vector, a grouping (character / factor) vector, a or a name of one in data. Ignored if x is a formula.

If names (as opposed to character vectors of names) are provided for x and y, an error is raised.  (Since the tidyverse, the trend is that actual names are allowed, which is what confused me at first, but I noticed later that the formula option indirectly allows for actual names.)

**Comment by mattansb** [comment]
Good catch!
I have fix the documentation for now...

However, I see there was an attempt to use actual names by @strengejacke here:
https://github.com/easystats/effectsize/blob/bb2754abe7a80575cbfac391571587dfbb645d2a/R/sd_pooled.R#L23

If Daniel is up for the task (when he gets back for holiday!), this might also be applicable in `cohens_d()`, `hedges_g()` and `glass_delta()`.

**Comment by strengejacke** [comment]
I planned to add NSE support, however, it turned out to require changes in so many other places, that I decided to revert everything. But I think we should (or do already?) support names as character vector.

**Comment by mattansb** [comment]
Yes, names as character vectors are supported.

**Comment by david-romano** [comment]
> I have fix the documentation for now...

Thanks for the super-quick response, @mattansb !

**Comment by mattansb** [comment]
I'm closing this for now. I think we offer enough ways to specify the data in these functions (and changing this requires a lot of code re-write).

**Comment by ASKurz** [comment]
The documentation for **effectsize** 0.4.5 describes the `y` argument for `sd_pooled()` as "A numeric vector, a grouping (character / factor) vector." However, `sd_pooled()` only seems to work when `y` is a factor. For example,

```{r}
library(effectsize)

dat <- data.frame(
  g = rep(c("treatment", "control"), each = 100),
  y = c(rnorm(n = 200))
)

sd_pooled(dat$y, factor(dat$g))
sd_pooled(dat$y, dat$g)
```

Whereas `sd_pooled(dat$y, factor(dat$g))` works just fine, `sd_pooled(dat$y, dat$g)` returns an error. It's be really great if both worked.

**Comment by mattansb** [comment]
Wow, that is very weird!
I'll fix this before the next release. Thanks!

**Comment by mattansb** [comment]
``` r
library(effectsize)

dat <- data.frame(
  g = rep(c("treatment", "control"), each = 100),
  y = c(rnorm(n = 200))
)

sd_pooled(dat$y, factor(dat$g))
#> [1] 1.011656
sd_pooled(dat$y, dat$g)
#> [1] 1.011656
sd_pooled(y ~ g, data = dat)
#> [1] 1.011656
sd_pooled(y ~ factor(g), data = dat)
#> [1] 1.011656
sd_pooled(dat$y ~ dat$g)
#> [1] 1.011656
sd_pooled(dat$y ~ factor(dat$g))
#> [1] 1.011656
sd_pooled("y", "g", data = dat)
#> [1] 1.011656
sd_pooled("y", dat$g, data = dat)
#> [1] 1.011656
sd_pooled(dat$y, "g", data = dat)
#> [1] 1.011656
```

<sup>Created on 2021-07-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by mattansb** [comment]
closed in #362
